spec.md

```diff
@@ -400,10 +407,45 @@ The following rules apply if `parameters` is included anywhere in the catalog:
 * Platforms SHOULD be prepared to support later versions of JSON schema.
 * The `$schema` key MUST be present in the schema declaring the version of JSON
 schema being used.
-* Schemas MUST NOT contain any external references.
+* Schemas MUST NOT contain any external references that are resolved to a URI
+  that requires additional fetching.
 * Schemas MUST NOT be larger than 64kB.
 
 
+###### $ref Usage
+
+It is likely that a Broker will have redundant definitions of `parameters` in
+the catalog. To simplfy this, the broker MAY provide a `root_schemas` entry. The
+Platform will need to process the catalog's `root_schemas` to fully qualify
+`$ref` object schema definitions.
+
+
+##### Root Schema Objects
+
+| Response field | Type | Description |
+| --- | --- | --- |
+| $schema* | string | The JSON Schema declaring the version of JSON |
+schema being used. |
+| $id* | string | The JSON Schema id to be referenced from elsewhere in the catalog. |
+| definitions* | Dictonary-of-JSON Schema Object | A definition of a JSON Schema Object. |
+
+The following rules apply if `root_schemas` is included anywhere in the catalog:
+* Platforms MUST support at least
+[JSON Schema draft v4](http://json-schema.org/).
+* Platforms SHOULD be prepared to support later versions of JSON schema.
+* The `$schema` key MUST be present in the schema declaring the version of JSON
+schema being used.
+* Schemas MUST NOT contain any external references that are resolved to a URI
+  that requires additional fetching.
+* `$id` MUST be a unique URI in the root schemas array.
+* All `definitions` MUST define a unique `$id` for the Root Schema Object in
```

Comment:
unique within that definition or across all?

Reply:
just the definition. 

the first point applies to the `root_schema`, the second point applies to the `definitions` object, `root_schemas` can have `definitions` that conflict as long as their parent `$id`s are unique.

Reply:
let's mention the scope in the doc.